mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp

```diff
@@ -198,6 +198,195 @@ class NvInferExecContextWrapper {
+  OutputAllocatorImpl &operator=(OutputAllocatorImpl &&) = delete;
+
+  void setTensorName(const std::string &tensorName) {
+    mTensorName = tensorName;
+  }
+  void setCurrentMemory(void *currentMemory, int64_t size) {
+    mCurrentMemory = currentMemory;
+    mOutputSize = size;
+  }
+
+  // IOutputAllocator interface implementation
+  void *reallocateOutputAsync(const char *tensorName, void *currentMemory,
+                              uint64_t size, uint64_t alignment,
+                              cudaStream_t stream) override {
+    assert(currentMemory == mCurrentMemory && "Output buffer mismatch");
+    assert(tensorName == mTensorName && "Tensor name mismatch");
+    assert(!mReallocateOutputCalled && "Duplicate call to reallocateOutput");
+    mReallocateOutputCalled = true;
+
+    size = std::max(size, static_cast<uint64_t>(1));
+
+    if (size > mOutputSize) {
+      size = roundUp(size, alignment);
+
+      if (mOutputPtr) {
+        cudaFree(mOutputPtr);
+      }
+
+      mOutputPtr = nullptr;
+      mOutputSize = 0;
+
+      void *newMemory;
+      if (cudaMalloc(&newMemory, size) == cudaSuccess) {
+        mOutputPtr = newMemory;
+        mOutputSize = size;
+      }
+      return mOutputPtr;
+    }
+    return mCurrentMemory;
+  }
+
+  void notifyShape(const char *tensorName,
+                   const nvinfer1::Dims &dims) override {
+    assert(mReallocateOutputCalled &&
+           "TensorRT must invoke reallocateOutput first");
+    assert(!mNotifyShapeCalled && "Duplicate call to notifyShape");
+    assert(tensorName == mTensorName && "Tensor name mismatch");
+    mNotifyShapeCalled = true;
+    mOutputDims = dims;
+  }
+
+  // Accessor methods
+  const nvinfer1::Dims &getOutputDims() const { return mOutputDims; }
+  bool isReallocateOutputCalled() const { return mReallocateOutputCalled; }
+  bool isNotifyShapeCalled() const { return mNotifyShapeCalled; }
+  void *getOutputPtr() const { return mOutputPtr; }
+  uint64_t getOutputSize() const { return mOutputSize; }
+  const std::string &getTensorName() const { return mTensorName; }
+
+private:
+  /// Round up the given value to the nearest multiple of n.
+  static inline uint64_t roundUp(uint64_t m, uint64_t n) {
+    return ((m + n - 1) / n) * n;
+  }
+
+  void *mOutputPtr{nullptr};
+  uint64_t mOutputSize{0};
+  bool mReallocateOutputCalled{false};
+  bool mNotifyShapeCalled{false};
+  nvinfer1::Dims mOutputDims;
+  std::string mTensorName;
+  void *mCurrentMemory{nullptr};
+};
+
+/// OutputAllocator - Manages multiple OutputAllocatorImpl instances.
+class OutputAllocator {
+public:
+  explicit OutputAllocator(int64_t nbResults);
+
+  // Disable copy and move operations
+  OutputAllocator(const OutputAllocator &) = delete;
+  OutputAllocator &operator=(const OutputAllocator &) = delete;
+  OutputAllocator(OutputAllocator &&) = delete;
+  OutputAllocator &operator=(OutputAllocator &&) = delete;
+
+  void registerAllocator(size_t index, const char *name, void *ptr,
```

Comment:
What is this method for? It doesn't seem to be used?

Reply:
This and all the below methods are used. There was a bad merge. I will update the PR with those changes.